compiler/src/dotty/tools/dotc/transform/localopt/ConstantFold.scala

```diff
@@ -21,18 +21,19 @@ import Simplify.desugarIdent
  *    out (nested) if with equivalent branches wrt to isSimilar. For example:
  *      - if (b) exp else exp → b; exp
  *      - if (b1) e1 else if (b2) e1 else e2 → if (b1 || b2) e1 else e2
+ *      - if(!b) e1 else e2 → if(b) e2 else e1
  *
  *  - Constant propagation over pattern matching.
  *
- *  @author DarkDimius, OlivierBlanvillain
+ *  @author DarkDimius, OlivierBlanvillain, gan74
  */
  class ConstantFold(val simplifyPhase: Simplify) extends Optimisation {
   import ast.tpd._
 
   def visitor(implicit ctx: Context) = NoVisitor
   def clear(): Unit = ()
 
-  def transformer(implicit ctx: Context): Tree => Tree = { x => preEval(x) match {
```

Comment:
I guess without `preEval` you could remove the `x => x match` altogether to turn this into a partial function:

```scala
def transformer(implicit ctx: Context): Tree => Tree = {
  case If(,,) => ...
}
```